src/simulators/density_matrix/densitymatrix.hpp

```diff
@@ -95,6 +95,9 @@ class DensityMatrix : public UnitaryMatrix<data_t> {
   // The matrix is input as vector of the matrix diagonal.
   void apply_diagonal_superop_matrix(const reg_t &qubits, const cvector_t<double> &mat);
 
+  //batched matrix is used for GPU optimization
+  void apply_batched_matrix(std::vector<batched_matrix_params>& params,reg_t& qubits,std::vector<std::complex<double>>& matrices){}
```

Comment:
Currently most of functions needed for batched multi-shots optimization is implemented for GPU only. So there are many empty functions to prevent compile errors.